### ISSUE TITLE
Fix creation of new entries

### ISSUE THREAD

**Issue by raphink** [camptocamp/puppet-puppetserver#14]
4a9d55a7561e67e0fdb593e7305ba21f1bcdbe7d introduced a new tree for the lens, which types every setting.

The manifests were fixed accordingly, but non-existing entries can not be created anymore.

For example, when passing:

```puppet
puppetserver::config::helper { 'webserver/port':
      setting => 'webserver.conf/@hash[.="webserver"]/@simple[.="port"]',
      type    => 'puppetserver',
}
```

it will fail if there is no entry for `port` yet.

The solution is to create each level with Augeas, e.g. in this case:

```puppet
$changes = [
  'set webserver.conf/@hash[.="webserver"] "webserver"',
  'set webserver.conf/@hash[.="webserver"]/@simple[.="port"] "port"',
  "set webserver.conf/@hash[.="webserver"]/@simple[.="port"]/@value '#{value}'",
]
```

And if we're going to do that, we might as well revert to the old interface, and add a `type` parameter.

**Comment by raphink** [comment]
So now we have a `puppetserver_config` type and provider, we "just" need to wire it properly to the manifests so that it works in a transparent way :-)

**Comment by raphink** [comment]
Fixed in 2.0.0
